pkgs/development/tools/marksman/default.nix

```diff
@@ -0,0 +1,50 @@
+{ lib
+, fetchFromGitHub
+, buildDotnetModule
+, dotnetCorePackages
+}:
+
+buildDotnetModule rec {
+  pname = "marksman";
+  version = "2022-09-13";
+
+  src = fetchFromGitHub {
+    owner = "artempyanykh";
+    repo = "marksman";
+    rev = version;
+    sha256 = "sha256-87XuapUeeIw+MQ8rosiG7p6SVNOmflBsb3CcvfaUdCU=";
+  };
+
+  patches = [ ./patches/0001-revert-embed-git-commit-into-the-version.patch ];
+
+  projectFile = "Marksman/Marksman.fsproj";
+
+  doCheck = true;
+  testProjectFile = "Tests/Tests.fsproj";
+
+  nugetDeps = ./deps.nix;
+
+  dotnet-sdk = dotnetCorePackages.sdk_6_0;
+  dotnet-runtime = dotnetCorePackages.runtime_6_0;
+
+  meta = with lib; {
+    description = "Language Server for Markdown";
+    longDescription = ''
+      Marksman is a program that integrates with your editor
+      to assist you in writing and maintaining your Markdown documents.
+      Using LSP protocol it provides completion, goto definition,
+      find references, rename refactoring, diagnostics, and more.
+      In addition to regular Markdown, it also supports wiki-link-style
+      references that enable Zettelkasten-like note taking.
+    '';
+    homepage = "https://github.com/artempyanykh/marksman";
+    license = licenses.mit;
```

Comment:
MIT requires:

> The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.

Currently this isn't copied into the package:

```console
tlater ~/result $ ls
bin  lib
tlater ~/result $ fd LICENSE .
```

Arguably a problem with their build scripts, I suppose, but let's copy it in, given their license asks for it? Something like

```nix
postInstall = ''
  mkdir -p $out/share/doc
  install -m 0644 LICENSE $out/share/doc/
'';
```

This isn't always upheld in nixpkgs, but it's a tiny amount of effort, so...

Reply:
> This isn't always upheld in nixpkgs

Yes, because it creates conflicts between packages when installed globally.